src/TMap.cpp

```diff
@@ -1028,8 +1028,17 @@ bool TMap::findPath(int from, int to)
     return false;
 }
 
-bool TMap::serialize(QDataStream& ofs)
+bool TMap::serialize(QDataStream& ofs, int saveVersion)
 {
+    // clamp version values
+    if (saveVersion < 0) saveVersion = 0;
+    else if (saveVersion > mMaxVersion) saveVersion = mMaxVersion;
```

Comment:
Are you sure clamping silently is a good idea - if say my script wanted to save in map format 15 and I run it on my Mudlet which supports it, it works fine. I give the script to my buddy who runs an old Mudlet but I don't know and his can only save in 13 and so it'll save in 13 and lose data - we'll then be confused as to what is going on

Reply:
`mMaxVersion` will never be lower than what he can read the map, so in your example you should save it with the 13 format before passing it to it. it's mostly there to prevent people from saving with a format that their version of mudlet does not known how to serialize

Reply:
How would I know I need to save it with 13?

Reply:
you pass you 15 to your friend, he tries to load it and mudlet tells him that the map `file format (15) is higher that this version of Mudlet can handle (13)!` He tells you he can only open 13 so you know you need to save it with 13

Reply:
There's people who'll miss that message and there will definitely be people who have no idea what it means, so they'll try saving it anyway thinking they're saving with 15 and they'll get 13 instead. My experience making systems says this will happen 100%.

So clamping is not really safe, I'd return nil + msg and also have a function to check for the max save version supported, then you can make a script that deals with all situations.

Reply:
that scenario will not happen because that line is already an error, and they will notice the map they are trying to load is not working, but I can see why a giving a message is nice.. I'll see to that